### PR TITLE
script: check GOPATH is set and install gogoproto if missing

### PR DESCRIPTION
**Pull request by hhkbp2**
Hi there,

This PR try to add these into the building of go code:

1. check GOPATH is set
2. download gogoproto code and install its binary locally if it's missing

They would be helpful to make the building process easier.

### CODE REVIEW

**Comment by c4pt0r** [comment]
LGTM

File: generate_go.sh [hunk 1]
```diff
@@ -1,11 +1,39 @@
...
+# download gogproto code and install its binary if it's missing
+if ! cmd_exists protoc-gen-gofast || [ ! -e "$GOGO_ROOT" ]; then
+    echo "gogoproto code/generator missing, try to download/install it"
+    go get ${gogo_protobuf_url}/proto
+    go get ${gogo_protobuf_url}/protoc-gen-gofast
+    go get ${gogo_protobuf_url}/gogoproto
+fi
+
+# if the gogoproto binary is installed locally, use it as the generator
+gogo_proto_bin=$PWD/bin/protoc-gen-gofast
```
Comment (siddontang) [inline generate_go.sh]: how can we install gogo in local bin?

Comment (hhkbp2) [inline generate_go.sh]: What's the problem?

Comment (siddontang) [inline generate_go.sh]: The common way is to install gogo in $GOPATH/bin, why should we consider local bin here?

Comment (hhkbp2) [inline generate_go.sh]: It's possible that the user set GOPATH to PWD(a common use case), and then run this script. If he does, it still works without modifying this build script.

Comment (siddontang) [inline generate_go.sh]: we can use  $GOPATH/bin directly, not $PWD/bin

Comment (hhkbp2) [inline generate_go.sh]: Address comment.

**Comment by siddontang** [comment]
LGTM

**Pull request merged**